dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/boogie/PthreadsProcedures.java

```diff
@@ -42,7 +42,8 @@ public static void handlePthreadsFunctions(VisitorBoogie visitor, Call_cmdContex
 			pthread_create(visitor, ctx);
 			break;
 		case "pthread_join":
-			pthread_join(visitor, ctx);
+			// VisitorBoogie already took care of creating the join event
+			// when it parsed the previous load.
```

Comment:
How does this work now? If the load is generated by Clang only, then where do we make sure that it is an acquire load?

Reply:
When visiting `pthread_create` in `PthreadsProcedures`, we add certain expressions to `visitor.allocations`. When creating a Load in `VisitoBoogie`, we check if the value is one of such expressions and create a Load or a Join event accordingly
```
if(allocations.contains(value)) {
    // These loads corresponding to pthread_joins
    child = EventFactory.Pthread.newJoin(register, (IExpr)value);
 else {
    child = EventFactory.newLoad(register, (IExpr)value, "");
}
```
With the code before this PR, we ended up creating two Join events, the one from the code above and the one from the code which this PR removes in `PthreadsProcedures`.